src/app/lib/analyticsUtils/index.js

```diff
@@ -333,4 +333,272 @@ export const getCampaignType = () => {
+    {
+      description: 'at_campaign field',
+      value: getMarketingUrlParam(query, 'at_campaign'),
+      wrap: false,
+    },
+    {
+      description: 'at_creation field',
+      value: getMarketingUrlParam(query, 'at_creation'),
+      wrap: true,
+    },
+    {
+      description: 'at_send_date field',
+      value: getMarketingUrlParam(query, 'at_send_date'),
+      wrap: false,
+    },
+    {
+      description: 'at_link field',
+      value: getMarketingUrlParam(query, 'at_link'),
+      wrap: true,
+    },
+    {
+      description: 'at_recipient_id + @ + at_recipient_list field',
+      value: hasRecipientFields
+        ? `${getMarketingUrlParam(
+            query,
+            'at_recipient_id',
+          )}@${getMarketingUrlParam(query, 'at_recipient_list')}`
+        : '',
+      wrap: false,
+    },
+  ]);
+};
+
+export const getDisplayMarketingString = href => {
+  const { query } = new Url(href, true);
+  return buildMarketingString([
+    {
+      description: 'display campaign prefix',
+      value: 'AD',
+      wrap: false,
+    },
+    {
+      description: 'at_campaign field',
+      value: getMarketingUrlParam(query, 'at_campaign'),
+      wrap: false,
+    },
+    {
+      description: 'at_creation field',
+      value: getMarketingUrlParam(query, 'at_creation'),
+      wrap: true,
+    },
+    {
+      description: 'at_variant field',
+      value: getMarketingUrlParam(query, 'at_variant'),
+      wrap: true,
+    },
+    {
+      description: 'at_format field',
+      value: getMarketingUrlParam(query, 'at_format'),
+      wrap: true,
+    },
+    {
+      description: 'blank value (-)',
+      value: '',
+      wrap: false,
+    },
+    {
+      description: 'at_general_placement field',
+      value: getMarketingUrlParam(query, 'at_general_placement'),
+      wrap: true,
+    },
+    {
+      description: 'at_detail_placement field',
+      value: getMarketingUrlParam(query, 'at_detail_placement'),
+      wrap: true,
+    },
+  ]);
+};
+
+export const getCustomMarketingString = href => {
+  const { query } = new Url(href, true);
+  return buildMarketingString([
+    {
+      description: 'custom campaign prefix',
+      value: `CS${getMarketingUrlParam(query, 'at_medium').replace(
+        'custom',
+        '',
+      )}`,
+      wrap: false,
+    },
+    {
+      description: 'at_campaign field',
+      value: getMarketingUrlParam(query, 'at_campaign'),
+      wrap: false,
+    },
+    {
+      description: 'at_custom1 field',
+      value: getMarketingUrlParam(query, 'at_custom1'),
+      wrap: true,
+    },
+    {
+      description: 'at_custom2 field',
+      value: getMarketingUrlParam(query, 'at_custom2'),
+      wrap: true,
+    },
+    {
+      description: 'at_custom_3 field',
+      value: getMarketingUrlParam(query, 'at_custom3'),
+      wrap: true,
+    },
+    {
+      description: 'at_custom_4 field',
+      value: getMarketingUrlParam(query, 'at_custom4'),
+      wrap: true,
+    },
+  ]);
+};
```

Comment:
I think it would be good to break this file up so the different marketing strings are in their own files? Maybe a quick refactor PR after this one?

Reply:
@andrewscfc I was thinking since this is ati specific, might be worth having most of this all namespaced somewhere within the [ati folder](https://github.com/bbc/simorgh/tree/latest/src/app/containers/ATIAnalytics) and not necessarily in a generic analytic folder but would welcome thoughts from others on this

Reply:
Sounds good, just good to follow SOLID with this stuff 👍

Reply:
I'll create an issue to cater for this refactor